torchio/transforms/transform.py

```diff
@@ -64,35 +64,39 @@ def __call__(
         """Transform data and return a result of the same type.
 
         Args:
-            data: Instance of :py:class:`~torchio.Subject`, 4D
-                :py:class:`torch.Tensor` or 4D NumPy array with dimensions
+            data: Instance of 1) :py:class:`~torchio.Subject`, 4D
+                :py:class:`torch.Tensor` or NumPy array with dimensions
                 :math:`(C, W, H, D)`, where :math:`C` is the number of channels
                 and :math:`W, H, D` are the spatial dimensions. If the input is
-                a tensor, the affine matrix is an identity and a tensor will be
-                also returned.
+                a tensor, the affine matrix will be set to identity. Other
+                valid input types are a SimpleITK image, a
+                :py:class:`torch.Image`, a NiBabel Nifti1 Image or a Python
+                dictionary. The output type is the same as te input type.
         """
         if torch.rand(1).item() > self.probability:
             return data
-
         data_parser = DataToSubject(data, keys=self.keys)
         subject = data_parser.get_subject()
-
         if self.copy:
             subject = copy.copy(subject)
-
         with np.errstate(all='raise'):
             transformed = self.apply_transform(subject)
-
         self.add_transform_to_subject_history(transformed)
-
         for image in transformed.get_images(intensity_only=False):
             ndim = image[DATA].ndim
             assert ndim == 4, f'Output of {self.name} is {ndim}D'
-
         output = data_parser.get_output(transformed)
-
         return output
 
+    def __repr__(self):
+        if hasattr(self, 'args_names'):
+            names = self.args_names
+            args_strings = (f'{arg}={getattr(self, arg)}' for arg in names)
+            args_string = ', '.join(args_strings)
+            return f'{self.name}({args_string})'
+        else:
+            return super().__repr__()
+
```

Comment:
just for my understanding, when is this __repr__ called ?

Reply:
When you print a transform.

```python
In [1]: import torchio as tio
   ...: colin = tio.datasets.Colin27()
   ...: transform = tio.Compose([tio.RescaleIntensity(), tio.RandomFlip(), tio.RandomBlur(), tio.CropOrPad((100, 200, 300))])
   ...: transformed = transform(colin)

In [2]: transformed.get_composed_history()
Out[2]: 
Compose(
    RescaleIntensity(out_min_max=(0, 1), percentiles=(0, 100), masking_method=<function NormalizationTransform.ones at 0x7f4da759bdc0>)
    Flip(axes=(0,))
    Blur(std={'t1': [tensor([0.2149, 1.7382, 1.0895])]})
    Pad(padding=(0, 0, 0, 0, 60, 59), padding_mode=constant)
    Crop(cropping=(41, 40, 9, 8, 0, 0))
)

In [3]: transformed.get_inverse_transform()
/home/fernando/git/torchio/torchio/transforms/augmentation/composition.py:56: UserWarning: Skipping RescaleIntensity as it is not invertible
  warnings.warn(message, UserWarning)
/home/fernando/git/torchio/torchio/transforms/augmentation/composition.py:56: UserWarning: Skipping Blur as it is not invertible
  warnings.warn(message, UserWarning)
Out[3]: 
Compose(
    Pad(padding=(41, 40, 9, 8, 0, 0), padding_mode=constant)
    Crop(cropping=(0, 0, 0, 0, 60, 59))
    Flip(axes=(0,))
)
```

I still need to figure out if there's a nice way to do this with random transforms.